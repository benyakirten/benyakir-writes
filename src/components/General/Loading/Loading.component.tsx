import * as React from 'react'

import { LoadingContainer } from './Loading.styles'

const Loading: React.FC<LoadingProps> = ({ size = '2rem' }) => (
  <LoadingContainer size={size} />
)

export default Loading
