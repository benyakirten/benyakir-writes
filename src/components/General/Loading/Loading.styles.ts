import styled from 'styled-components'

import { ABS_BLACK } from '@StyleVars'

export const LoadingContainer = styled.span<{ size: string }>`
  display: inline-block;

  width: ${(props) => props.size};
  height: ${(props) => props.size};
  border-radius: 50%;
  border: 2px solid #ddd;
  border-top: 1px solid ${(props) => props.theme.base.textColor};

  animation: rotate 1s infinite linear;

  @keyframes rotate {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
`
