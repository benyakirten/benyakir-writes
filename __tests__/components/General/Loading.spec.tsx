import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer'

import Loading from "@Gen/Loading/Loading.component"

describe('Loading component', () => {
    afterEach(cleanup)
    
    it('should render correctly', () => {
        expect(() => render(<Loading />)).not.toThrow()

        const loading = renderer.create(<Loading />).toJSON()
        expect(loading).toMatchSnapshot()
    })
})