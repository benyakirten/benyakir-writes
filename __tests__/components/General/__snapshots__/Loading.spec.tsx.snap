// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render correctly 1`] = `
<span
  className="sc-bdvvtL kRglAr"
  size="2rem"
/>
`;
